packages/@guardian/source-react-components/scripts/fetch-icons.ts

```diff
@@ -0,0 +1,257 @@
+
+const vendorIconPath = path.resolve(__dirname, '..', 'vendor', 'icons');
+
+const FIGMA_API_OPTIONS = {
+	headers: {
+		'X-Figma-Token': process.env.FIGMA_TOKEN ?? 'ADD ME!!!',
+	},
+};
+
+// SVGR uses babel under the hood to create the jsx
+// this is our custom transform for it to use
+const createBabelConfig = (iconName: string) => {
+	const babelConfig: TransformOptions = {};
+	const plugins = [];
+
+	// remove fill attribute from all icons except those with special fill colours
+	if (!ICONS_WITH_FILL_COLOURS.includes(iconName)) {
+		plugins.push([
+			'@svgr/babel-plugin-remove-jsx-attribute',
+			{
+				elements: ['svg', 'path'],
+				attributes: ['fill'],
+			},
+		]);
+	}
+
+	// replace viewbox with legacy 30x30 viewbox
+	// TODO: a future version of Source should expose icons with the viewboxes
+	// defined in Figma
+	plugins.push([
+		'@svgr/babel-plugin-replace-jsx-attribute-value',
+		{
+			values: [
+				{ value: '0 0 24 24', newValue: '-3 -3 30 30' },
+				// legacy 50x20 wide viewbox
+				{ value: '0 0 48 24', newValue: '-1 2 50 20' },
+			],
+		},
+	]);
+
+	if (plugins.length === 0) {
+		return {};
+	}
+
+	babelConfig.plugins = plugins;
+
+	return babelConfig;
+};
+
+// now we start fetching SVGs and building our components
+void (async () => {
+	// remove any existing icons from a previous run
+	try {
+		await rm(vendorIconPath, { recursive: true });
+	} catch (e) {
+		// do nothing
+	}
+
+	// create the icons directory
+	await mkdirp(vendorIconPath);
+
+	// create a readme
+	await writeFile(path.resolve(vendorIconPath, 'README.md'), readme, 'utf8');
+
+	// Get a list of available (figma) components from Figma
+	// https://www.figma.com/developers/api#library-items-types
+	const figmaComponents = (
+		await axios.get<FigmaComponentsResponse>(
+			`https://api.figma.com/v1/files/${ICON_FILE}/components`,
+			{
+				headers: {
+					'X-Figma-Token': process.env.FIGMA_TOKEN ?? 'ADD ME!!!',
+				},
+			},
+		)
+	).data.meta.components;
+
+	// filter out the icons from the list of figma components
+	const figmaIconComponents = figmaComponents.filter((c) => {
+		return (
+			// Only get icons that are in a certain frame
+			c.containing_frame &&
+			c.containing_frame.name === 'UI icons 24(w)x24(w)'
+		);
+	});
+
+	const iconIds = figmaIconComponents.map(({ node_id }) => node_id).join(',');
+
+	// Get the URLs we can fetch actual images from from Figma
+	// https://www.figma.com/developers/api#get-images-endpoint
+	const figmaIconSvgUrlsByNodeId = (
+		await axios.get<FigmaImagesResponse>(
+			`https://api.figma.com/v1/images/${ICON_FILE}/?ids=${iconIds}&format=svg`,
+			FIGMA_API_OPTIONS,
+		)
+	).data.images;
+
+	// fetch the SVGs from Figma, create a react component from them using svgr
+	// and save them both
+	for (const icon of figmaIconComponents) {
+		const url = figmaIconSvgUrlsByNodeId[icon.node_id];
+
+		// Fetch SVG markup from Figma
+		const svg = (await axios.get<string>(url)).data;
```

Comment:
I've found destructuring to be a nice alternative to complex responses, so thought I'd share:

```suggestion
		const { data: svg } = await axios.get<string>(url);
```